File: backend/src/main/java/com/ocs/service/DoctorService.java
package com.ocs.service;

import java.util.List;

import com.ocs.dto.ApiResponse;
import com.ocs.dto.AppointmentResponse;
import com.ocs.dto.DoctorDTO;
import com.ocs.dto.LeaveRequestDTO;

public interface DoctorService {

    // Get all doctors - used by Admin dashboard
    List<DoctorDTO> getAllDoctors();

    // Get logged-in doctor's profile
    DoctorDTO getDoctorByEmail(String email);

    // Get logged-in doctor's appointments
    List<AppointmentResponse> getDoctorAppointments(String email);

    // Toggle doctor availability
    ApiResponse toggleAvailability(String email);

    // Request doctor leave
    ApiResponse requestLeave(String email, LeaveRequestDTO leaveDTO);

    // Get logged-in doctor's leaves
    List<LeaveRequestDTO> getDoctorLeaves(String email);
}

File: frontend/src/services/doctorService.js
import api from './api';

export const doctorService = {
  getProfile: async () => {
    const res = await api.get('/doctor/profile');
    return res.data;
  },

  getAppointments: async () => {
    const res = await api.get('/doctor/appointments');
    return res.data;
  },

  toggleAvailability: async () => {
    const res = await api.put('/doctor/availability/toggle');
    return res.data;
  },

  requestLeave: async (leaveData) => {
    const res = await api.post('/doctor/leaves', leaveData);
    return res.data;
  },

  getLeaves: async () => {
    const res = await api.get('/doctor/leaves');
    return res.data;
  }
};
